app/models/application_draft.rb

```diff
@@ -69,8 +71,19 @@ def ready?
     false # valid?(:apply)
   end
 
-  def state
-    (applied_at? ? 'applied' : 'draft').inquiry
+  aasm :column => :state, :no_direct_assignment => true do
+    state :draft, :initial => true
+    state :applied
+
+    event :submit_application do
+      after do |applied_at_time = nil|
+        self.applied_at = applied_at_time || Time.now
+      end
+
+      # If the transition should only happen if the object is `ready?`,
+      # add a :guard => :ready? here
+      transitions :from => :draft, :to => :applied
```

Comment:
Can you add a test that checks that one can / cannot transition to `applied` with stubbed return values of `ready?` ?